controllers/certificatesigningrequests_controller.go

```diff
@@ -0,0 +1,170 @@
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.7.2/pkg/reconcile
+func (r *certificateSigningRequestsReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = r.log.WithValues("certificatesigningrequests", req.NamespacedName)
+
+	certificateSigningRequest := &certificatesv1.CertificateSigningRequest{}
+	if err := r.client.Get(ctx, req.NamespacedName, certificateSigningRequest); err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	configMap, err := r.k8sclientset.CoreV1().ConfigMaps(r.watchNamespace).Get(ctx, InstanceConfigMap, metav1.GetOptions{})
+	if err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, r.reconcileCSR(certificateSigningRequest, configMap)
+}
+
+// NewCertificateSigningRequestsReconciler returns a pointer to certificateSigningRequestsReconciler
+func NewCertificateSigningRequestsReconciler(mgr manager.Manager, watchNamespace string) (*certificateSigningRequestsReconciler, error) {
+	k8sClientSet, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating kubernetes clientset")
+	}
+
+	return &certificateSigningRequestsReconciler{
+		client:         mgr.GetClient(),
+		log:            ctrl.Log.WithName("controllers").WithName("CertificateSigningRequests"),
+		k8sclientset:   k8sClientSet,
+		watchNamespace: watchNamespace,
+	}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *certificateSigningRequestsReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	certificateSigningRequestsPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return pendingCertFilter(e.Object)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return pendingCertFilter(e.ObjectNew)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return pendingCertFilter(e.Object)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool { return false },
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&certificatesv1.CertificateSigningRequest{}, builder.WithPredicates(certificateSigningRequestsPredicate)).
+		Complete(r)
+}
+
+// pendingCertFilter returns true if the CSR is pending approval
```

Comment:
The comment of this function sounds really similiar to the isPending() function. Could we call out the difference?